backend/src/pong/pong.service.ts

```diff
@@ -0,0 +1,156 @@
+import { Injectable } from '@nestjs/common';
+import { Ball, GameState, MovementKeys, Paddle, Position } from './Game.types';
+
+@Injectable()
+export class PongService {
+	private readonly paddleWidth = 1;
+	private readonly paddleHeight = 12;
+
+	private readonly ballRadius = 1;
+
+	// ball direction and speed %
+	private dx = -0.4;
+	private dy = 0.4;
+
+	createNewGameState(): GameState {
+		const newGameState: GameState = {
+			playerOnePaddle: {
+				position: {
+					x: 0,
+					y: 50,
+				},
+				width: this.paddleWidth,
+				height: this.paddleHeight,
+			},
+
+			playerTwoPaddle: {
+				position: {
+					x: 100,
+					y: 50,
+				},
+				width: this.paddleWidth,
+				height: this.paddleHeight,
+			},
+
+			ball: {
+				position: {
+					x: 40,
+					y: 65,
+				},
+				radius: this.ballRadius,
+			},
+			canvas: {
+				x: 1200,
+				y: 960,
+			},
+		};
+		return newGameState;
+	}
+
+	doesHitWall(radius: number, pos: number, direction: number): boolean {
+		const fullHeigthOrWidth = 100;
+		return (
+			pos + direction > fullHeigthOrWidth - radius || pos + direction < radius
+		);
+	}
+
+	doesHitPaddle(ball: Ball, rect: Paddle, canvas: Position) {
+		const paddleY = rect.position.y;
+		const paddleHeight = rect.height;
+		const paddleTop = paddleY - paddleHeight / 2;
```

Comment:
is this paddle centre than? because of the /2?

Reply:
not sure where you start counting with y but it sounds like the exact opposite haha but if this works its fine

Reply:
count starts on top left corner, x = 0, y = 0, so right bottom corner x = width and y = height.
So the lower y becomes for a element the higher up it'll be on the canvas :)
It works exactly opposite to how we are used to counting cause y - 1 is up, and y + 1 is down.